bundles/org.openhab.binding.samsungtv/src/main/java/org/openhab/binding/samsungtv/internal/service/RemoteControllerService.java

```diff
@@ -48,203 +43,124 @@
+                : upnp ? supportedCommandsUpnp : supportedCommandsNonUpnp);
+        if (getArtModeSupported()) {
+            supported.addAll(refresh ? refreshArt : supportedCommandsArt);
+        }
+        if (getArtMode2022()) {
+            supported.addAll(refresh ? Arrays.asList() : art2022);
+        }
+        if (remoteController.noApps() && getPowerState() && refresh) {
+            supported.addAll(refreshApps);
+        }
+        if (!refresh) {
+            logger.trace("{}: getSupportedChannelNames: {}", host, supported);
         }
-        logger.trace("getSupportedChannelNames: {}", supported);
         return supported;
     }
 
     @Override
-    public void addEventListener(EventListener listener) {
-        listeners.add(listener);
-    }
-
-    @Override
-    public void removeEventListener(EventListener listener) {
-        listeners.remove(listener);
-    }
-
     public boolean checkConnection() {
-        if (remoteController != null) {
-            return remoteController.isConnected();
-        } else {
-            return false;
-        }
+        return remoteController.isConnected();
     }
 
     @Override
     public void start() {
-        if (remoteController != null) {
-            try {
-                remoteController.openConnection();
-            } catch (RemoteControllerException e) {
-                logger.warn("Cannot open remote interface ({})", e.getMessage());
-            }
-            return;
-        }
-
-        String protocol = (String) getConfig(SamsungTvConfiguration.PROTOCOL);
-        logger.info("Using {} interface", protocol);
-
-        if (SamsungTvConfiguration.PROTOCOL_LEGACY.equals(protocol)) {
-            remoteController = new RemoteControllerLegacy(host, port, "openHAB", "openHAB");
-        } else if (SamsungTvConfiguration.PROTOCOL_WEBSOCKET.equals(protocol)
-                || SamsungTvConfiguration.PROTOCOL_SECUREWEBSOCKET.equals(protocol)) {
-            try {
-                remoteController = new RemoteControllerWebSocket(host, port, "openHAB", "openHAB", this);
-            } catch (RemoteControllerException e) {
-                reportError("Cannot connect to remote control service", e);
-            }
-        } else {
-            remoteController = null;
-            return;
-        }
-
-        if (remoteController != null) {
-            try {
+        try {
+            if (!checkConnection()) {
                 remoteController.openConnection();
-            } catch (RemoteControllerException e) {
-                reportError("Cannot connect to remote control service", e);
             }
+        } catch (RemoteControllerException e) {
+            reportError("Cannot connect to remote control service", e);
         }
+        previousApp = "";
     }
 
     @Override
     public void stop() {
-        if (remoteController != null) {
-            try {
+        try {
+            if (getArtMode2022()) {
+                artMode = false;
```

Comment:
Probably not anymore. Samsung keeps changing their firmware, so I think this was a fix for a change they made a while back, that they have since reverted.

I think I can remove it safely.